### PR TITLE
Remove levenshtein check from lint

### LINKED ISSUE #378: Name collision detection gives mostly false positives
The name collision detection lint mostly gives false positives. We should tune it down to be less eager.

As an example (frm https://github.com/ocaml/opam-repository/pull/26658) it is giving

    Warning in dblp-api.0.1.1: Possible name collision with package 'bap-api'
    Warning in dblp.0.1.1: Possible name collision with package 'rlp'
    Warning in dblp.0.1.1: Possible name collision with package 'nlp'
    Warning in dblp.0.1.1: Possible name collision with package 'lp'
    Warning in dblp.0.1.1: Possible name collision with package 'dmap'
    Warning in dblp.0.1.1: Possible name collision with package 'dlm'
    Warning in dblp.0.1.1: Possible name collision with package 'dbm'
    Warning in dblp.0.1.1: Possible name collision with package 'dbf'
    Warning in dblp.0.1.1: Possible name collision with package 'dap'
    Warning in dblp.0.1.1: Possible name collision with package 'bap'

Given that the intent is to alert to possible malicious punning, these warnings are totally off the mark.

### PR DESCRIPTION
**Pull request by shonfeder**
Closes #378

### CODE REVIEW

File: opam-ci-check/lib/lint.ml [hunk 1]
```diff
@@ -309,40 +309,18 @@ module Checks = struct
...
-    - [lru-cache] and [lru_cache] collide
-    - [lru-cache] and [LRU-cache] collide
-    - [lru-cache] and [cache-lru] do not collide
-    - [ocaml] and [pcaml] collide
-    - [ocamlfind] and [ocamlbind] do not collide *)
+      and dashes ([-]) to be equal *)
   let package_name_collision p0 p1 =
     let dash_underscore p0 p1 =
-      let f = function '_' -> '-' | c -> c in
+      let f = function '_' -> '-' | c -> Char.lowercase_ascii c in
```
Comment (punchagan) [inline opam-ci-check/lib/lint.ml:318]: Do we want to tweak this to account for `-/_/<nothing>` as per this [comment](https://github.com/ocurrent/opam-repo-ci/issues/378#issuecomment-2425389474)?

**Review by punchagan — APPROVED** [review]

File: opam-ci-check/lib/lint.ml (hunk 1 above)
Comment (shonfeder) [reply opam-ci-check/lib/lint.ml]: Good point. I added that logic in ddde7df and tests for it in 6975fb0 .

**Review by punchagan — APPROVED** [review]
This looks good to me! Thanks for taking care of this so quickly!

**Comment by shonfeder** [comment]
Thank you for the input and reviews!

**Pull request merged**